### DISCUSSION TITLE
Why not provide router implementations with the library?

### DISCUSSION THREAD

**Discussion by xxfast** [arkivanov/Decompose#85] [Q&A] [answered]
Hi @arkivanov. 

I've followed your article [here](https://proandroiddev.com/a-comprehensive-hundred-line-navigation-for-jetpack-desktop-compose-5b723c4f256e) to implement navigation on my compose desktop app. It works great 👍 Thank you for this amazing article.  I recently updated to 0.6.0 and had to make some changes to make it work again.

Is there a reason why these are not being served as a part of the `decompose.router` library?

**Comment by arkivanov** [comment, marked as answer]
Thanks for the feedback! The approach described in the article has some limitations - e.g. the nested navigation shouldn't work. But some of the reasons of why is it not included in the library:

- I always advocate to not mixing the business logic with UI
- And to write the business logic using plain classes, and use Compose for UI only (in case of writing apps)

Please also check the [main points](https://github.com/arkivanov/Decompose#why-decompose) of Decompose.

That article is there to demonstrate that Decompose is very flexible, and can work in many different use cases. But that particular approach is against the ideology.

**Reply by arkivanov** [reply]
There is another limitation with the approach described in the article - Composables in the back stack are destroyed. There is no way to scope things while a certain Composable is in the back stack. Decompose's components do have this feature, as well as `navigation-compose` and `ViewModels`.
